src/class/vendor/vendor_device.c

```diff
@@ -112,10 +114,11 @@ static bool maybe_transmit(vendord_interface_t* p_itf)
   TU_VERIFY( !usbd_edpt_busy(TUD_OPT_RHPORT, p_itf->ep_in) );
 
   uint16_t count = tu_fifo_read_n(&p_itf->tx_ff, p_itf->epin_buf, CFG_TUD_VENDOR_EPSIZE);
-  if (count > 0)
+  if (count > 0 || last_in_transfer_was_epsize)
   {
     TU_ASSERT( usbd_edpt_xfer(TUD_OPT_RHPORT, p_itf->ep_in, p_itf->epin_buf, count) );
   }
+  last_in_transfer_was_epsize = count && (count == CFG_TUD_VENDOR_EPSIZE);
```

Comment:
condition seems to be overkill (only right side of && makes a difference) or wrong (&& where || should be)

Reply:
`&&` ist correct, but you're right, it's redundant.